integer/src/lib.rs

```diff
@@ -665,6 +665,47 @@ impl_integer_for_usize!(u32, test_integer_u32);
 impl_integer_for_usize!(u64, test_integer_u64);
 impl_integer_for_usize!(usize, test_integer_usize);
 
+/// An iterator over binomial coefficients.
+pub struct BinomialCoeff<T> {
+    a: T,
+    n: T,
+    k: T,
+}
+
+impl<T> BinomialCoeff<T>
+    where T: Integer,
+{
+    /// For a given n, iterate over all binomial coefficients ((k, n - k), binomial(n, k)).
+    pub fn new(n: T) -> BinomialCoeff<T> {
+        BinomialCoeff {
+            k: T::zero(), a: T::one(), n: n
+        }
+    }
+}
+
+impl<T> Iterator for BinomialCoeff<T>
+    where T: Integer + Clone,
+          for<'a> &'a T: std::cmp::PartialEq<&'a T>
```

Comment:
I don't think you need this `PartialEq<&'a T>`.  We already get `PartialEq<T>` from `Integer`, and there are blanket implementations in the standard library for reference variants from there.